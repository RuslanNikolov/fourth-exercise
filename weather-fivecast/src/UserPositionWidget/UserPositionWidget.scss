.position-box {
    display: flex;
    width: 100%;
    padding: 30px;
    justify-content: center;
    align-items: center;
}

.positon-detail-column {
    display: flex;
    justify-content: space-between;
    align-items: center;
    flex-direction: column;
    margin: 20px;
}

.position-label {
   font-weight: bold;
}
