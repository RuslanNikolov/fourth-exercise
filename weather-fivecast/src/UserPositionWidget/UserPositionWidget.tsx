import { useState, useEffect } from 'react';
import { Box, Typography } from '@mui/material';
import './UserPositionWidget.scss';

interface IPositionState {
    latitude: number;
    longtitude: number;
}

const UserPositionWidget = () => {
    const [position, setPosition] = useState<IPositionState>({
        latitude: 0,
        longtitude: 0
    });

    // fetch the posiiton once
    useEffect(() => {
        navigator.geolocation.getCurrentPosition((position: GeolocationPosition) => {
            setPosition({
                latitude: position.coords.latitude,
                longtitude: position.coords.longitude
            })
        })
    }, []) 

    return (
        <Box className="position-box">
            <Box className="positon-detail-column">
                <Typography>Your Latitude:</Typography>
                <Typography>{position.latitude}</Typography>
            </Box>

            <Box className="positon-detail-column">
                <Typography>Your Longtitude:</Typography>
                <Typography>{position.longtitude}</Typography>
            </Box>

            <Box className="positon-detail-column">
                <Typography>Your Attitude:</Typography>
                <Typography>&#10003;</Typography>
            </Box>
        </Box>
    )
}

export default UserPositionWidget;