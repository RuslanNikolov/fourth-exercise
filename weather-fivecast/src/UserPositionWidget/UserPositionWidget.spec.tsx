import React from 'react';
import renderer from 'react-test-renderer';
import enzyme from 'enzyme';
import UserPositionWidget from './UserPositionWidget';

it('renders correctly', () => {
    const component = renderer.create(<UserPositionWidget />).toJSON();
    expect(component).toMatchSnapshot();
});