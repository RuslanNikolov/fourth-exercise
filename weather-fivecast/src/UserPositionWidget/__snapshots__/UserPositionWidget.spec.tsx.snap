// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="position-box MuiBox-root css-0"
>
  <div
    className="positon-detail-column MuiBox-root css-0"
  >
    <p
      className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
    >
      Your Latitude:
    </p>
    <p
      className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
    >
      0
    </p>
  </div>
  <div
    className="positon-detail-column MuiBox-root css-0"
  >
    <p
      className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
    >
      Your Longtitude:
    </p>
    <p
      className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
    >
      0
    </p>
  </div>
  <div
    className="positon-detail-column MuiBox-root css-0"
  >
    <p
      className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
    >
      Your Attitude:
    </p>
    <p
      className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
    >
      âœ“
    </p>
  </div>
</div>
`;
